### PR TITLE
Fixing infinite spinning animation on user menu ctrl+click

### PR DESCRIPTION
**Pull request by MorrisJobke**
- rebase of #652 

cc @Faldon

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@Faldon Does this look right? Could you check this out and retry?

What I did to get this:

* fetch latest master
* checkout your branch
* do a `git rebase master`
* there where two conflicts in core/js/js.js -> resolve them (both resulted in empty changeset and I needed to do a `git rebase --skip` to skip the resulting empty commits - this was told me by the git command line interface):

```
$ git rebase --continue 
Applying: Added OS X meta key check for stopping spinning animation in app and user menu.
No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

**Comment by MorrisJobke** [comment]
cc @icewind1991 @schiessle @jancborchardt @ChristophWurst @juliushaertl

**Comment by Faldon** [comment]
Checked out and it works as expected. Thanks for helping out. Have to practice the rebase a little bit more.

**Comment by schiessle** [comment]
looks good :+1:

**Comment by MorrisJobke** [comment]
The menu now closes on right click. So the context menu of the right click hang somewhere in the UI, but the right clicked element is not visible anymore. @Faldon Is this the wanted behaviour?

**Comment by Faldon** [comment]
I personally prefer it that way. Although it makes debugging a little bit harder.
But I understand that not everyone expects the menu closes on right click. So, it is the wanted behavior by the programmer, not necessarily the wanted behavior by the community. :-)

**Comment by jancborchardt** [comment]
It shouldn’t close on right-click, doesn’t do that at the moment either. ;) Let’s stick to fixing the one issue at hand and not putting other stuff in.

**Comment by Faldon** [comment]
Ok. I rewrote it to stay open and not showing the spinner animation when doing a right click on the menu.

File: core/js/js.js [hunk 1]
```diff
@@ -1501,32 +1501,49 @@ function initCore() {
...
 			}
-			if(!event.ctrlKey) {
+			if(event.which === 1 && !event.ctrlKey && !event.metaKey) {
 				$app.addClass('app-loading');
 			} else {
-				// Close navigation when opening app in
-				// a new tab
-				OC.hideMenus();
+				// On middle click or on first button click with ctrl key or meta key hold
+				console.log(event.which);
```
Comment (MorrisJobke) [inline core/js/js.js]: debug log? ;)

**Comment by Faldon** [comment]
Ach f* -.-
On the run atm, but will fix it this evening.

**Comment by MorrisJobke** [comment]
> Ach f* -.-On the run atm, but will fix it this evening.

No worries 😄

**Comment by juliushaertl** [comment]
works fine now :+1:

**Comment by juliushaertl** [comment]
@MorrisJobke can we merge this or is it required to have another reviewer, as there were some more changes after @schiessle gave his thumbs up?

**Comment by MorrisJobke** [comment]
> @MorrisJobke can we merge this or is it required to have another reviewer, as there were some more changes after @schiessle gave his thumbs up?

I will test this again and review it finally.

**Comment by MorrisJobke** [comment]
Tested and works now as wanted 👍

**Comment by MorrisJobke** [comment]
I would keep this as it is in `master` and not backport this. @jancborchardt is this okay for you?

**Comment by MorrisJobke** [comment]
This kills for me and @nickvergessen navigation between apps (for Joas all apps, for me only gallery -.-) in Firefox. I will revert this now.

**Comment by MorrisJobke** [comment]
Reverted with #929

@Faldon Could you create your PR again and only solve it step by step. I guess the issue here was the "open with middleclick" thing. For now it would be enough to first mimic the current apps menu behavior and fix the middle click in a separate PR.

**Comment by Faldon** [comment]
I will look into it.

**Comment by MorrisJobke** [comment]
> I will look into it.

Thanks a lot 👍

**Comment by Faldon** [comment]
@MorrisJobke  So, I created a new PR to have a clean base.

**Pull request merged**